tardis/montecarlo/montecarlo_numba/tests/test_base.py

```diff
@@ -3,18 +3,70 @@
 import os
 import numpy.testing as npt
 from copy import deepcopy
+from tardis.base import run_tardis
+from pandas.testing import assert_frame_equal
 
 from tardis.montecarlo import (
     montecarlo_configuration as montecarlo_configuration,
 )
 from tardis.simulation import Simulation
+from tardis.montecarlo.montecarlo_numba.base import obj_list_to_dataframe
+
+
+@pytest.fixture(scope="module")
+def simulation_rpacket_tracking_enabled(config_verysimple, atomic_dataset):
+    config_verysimple.montecarlo.iterations = 3
+    config_verysimple.montecarlo.no_of_packets = 4000
+    config_verysimple.montecarlo.last_no_of_packets = -1
+    config_verysimple.spectrum.virtual.virtual_packet_logging = True
+    config_verysimple.montecarlo.no_of_virtual_packets = 1
+    config_verysimple.montecarlo.tracking.track_rpacket = True
+    config_verysimple.spectrum.num = 2000
+    atomic_data = deepcopy(atomic_dataset)
+    sim = run_tardis(
+        config_verysimple,
+        atom_data=atomic_data,
+        show_convergence_plots=False,
+    )
+    return sim
 
 
 @pytest.mark.xfail(reason="To be implemented")
 def test_montecarlo_radial1d():
     assert False
 
 
+def test_obj_list_to_dataframe(simulation_rpacket_tracking_enabled):
+    sim = simulation_rpacket_tracking_enabled
+    df_to_check = obj_list_to_dataframe(sim.runner.rpacket_tracker)
+
+    # check df shape and column names
+    assert df_to_check.shape == (
+        sum([len(tracker.r) for tracker in sim.runner.rpacket_tracker]),
+        6,
+    )
+    assert npt.assert_array_equal(
+        sim.runner.rpacket_tracker_df.columns.values,
+        ["status", "r", "nu", "mu", "energy", "shell_id"],
+    )
+
+    # check all data with rpacket_tracker
+    rtracker_list = []
```

Comment:
Maybe rename to something like `expected_rtrackers` to specify this is the expected output against we're checking.